File: lib/covidtracer_migration/ios/CovidTracerMigration.m

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CovidTracerMigration, NSObject)

RCT_EXTERN_METHOD(applyPublicFileProtection:(NSString *)publicFolderPath resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(findPublicKey:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(findPrivateKey:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(readLastLaunchTime:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

@end

File: lib/covidtracer_migration/ios/CovidTracerMigration.swift

import Foundation

@objc(CovidTracerMigration)
class CovidTracerMigration: NSObject
{
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return true
    }

    @objc
    func applyPublicFileProtection(_ publicFolderPath: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        Repository.applyFileProtection(publicFolderPath: publicFolderPath)
        resolve(nil)
    }

    @objc
    func findPublicKey(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        let result = Repository.getPublicKey()?.base64EncodedString(options: [])
        resolve(result)
    }

    @objc
    func findPrivateKey(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        let result = Repository.getPrivateKey()?.base64EncodedString(options: [])
        resolve(result)
    }
}
